frappe/desk/doctype/workspace/workspace.py

```diff
@@ -64,6 +64,9 @@ def validate(self):
 		except Exception:
 			frappe.throw(_("Content data shoud be a list"))
 
+	def after_insert(self):
+		frappe.cache.hdel("bootinfo", self.for_user or frappe.session.user)
```

Comment:
Semantically all cache clearing should happen in `clear_cache` method (which gets called on changes)

So move this to `clear_cache`?

Example 
https://github.com/frappe/frappe/blob/c83b8b2bb674e7d23b2f710c52eefe686bf3c8b3/frappe/website/website_generator.py#L72-L74